tests/ttnn/unit_tests/gtests/tensor/test_distributed_tensor.cpp

```diff
@@ -0,0 +1,188 @@
+    Tensor concatenated_tensor = api::aggregate_tensor(sharded_tensor, *composer);
+
+    Tensor expected_tensor =
+        from_vector(test_data, GetTensorSpec(ttnn::SimpleShape{num_devices, 1, 3, 1}, DataType::FLOAT32));
+    EXPECT_TRUE(ttnn::allclose<float>(concatenated_tensor, expected_tensor));
+}
+
+TEST_F(TensorDistributionTest, Shard2DInvalidMeshShape) {
+    const auto [num_rows, num_cols] = mesh_device_->shape();
+    ASSERT_EQ(num_rows, 2);
+    ASSERT_EQ(num_cols, 4);
+
+    EXPECT_ANY_THROW(
+        api::shard_tensor_2d_to_mesh_mapper(*mesh_device_, MeshShape{3, 1}, Shard2dConfig{.row_dim = 1, .col_dim = 2}));
+
+    EXPECT_ANY_THROW(
+        api::shard_tensor_2d_to_mesh_mapper(*mesh_device_, MeshShape{2, 5}, Shard2dConfig{.row_dim = 1, .col_dim = 2}));
+}
+
+TEST_F(TensorDistributionTest, Shard2DInvalidShardConfig) {
+    EXPECT_ANY_THROW(api::shard_tensor_2d_to_mesh_mapper(*mesh_device_, MeshShape{2, 4}, Shard2dConfig{}));
+}
+
+TEST_F(TensorDistributionTest, Concat2DInvalidConfig) {
+    EXPECT_ANY_THROW(api::concat_mesh_2d_to_tensor_composer(*mesh_device_, Concat2dConfig{.row_dim = 2, .col_dim = 2}));
+}
+
+TEST_F(TensorDistributionTest, Shard2DReplicateDim) {
+    const auto [num_rows, num_cols] = mesh_device_->shape();
+    ASSERT_EQ(num_rows, 2);
+    ASSERT_EQ(num_cols, 4);
+    const int num_devices = num_rows * num_cols;
+
+    std::vector<float> test_data = {0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0};
+    Tensor input_tensor =
+        from_vector(test_data, GetTensorSpec(ttnn::SimpleShape{1, num_rows, num_cols, 1}, DataType::FLOAT32));
+    input_tensor.print();
+
+    auto mapper = api::shard_tensor_2d_to_mesh_mapper(
+        *mesh_device_,
+        MeshShape{num_rows, num_cols},
+        Shard2dConfig{
+            .row_dim = 1,
+        });
+    Tensor sharded_tensor = api::distribute_tensor(input_tensor, *mesh_device_, *mapper);
+    sharded_tensor.print();
+
+    std::vector<Tensor> device_tensors = api::get_device_tensors(sharded_tensor);
+    EXPECT_EQ(device_tensors.size(), mesh_device_->num_devices());
+
+    int i = 0;
+    for (; i < 4; i++) {
+        EXPECT_THAT(to_vector<float>(device_tensors[i]), ElementsAre(0.0, 1.0, 2.0, 3.0));
+    }
+    for (; i < device_tensors.size(); i++) {
+        EXPECT_THAT(to_vector<float>(device_tensors[i]), ElementsAre(4.0, 5.0, 6.0, 7.0));
+    }
+}
+
+TEST_F(TensorDistributionTest, Shard2D) {
+    const auto [num_rows, num_cols] = mesh_device_->shape();
+    ASSERT_EQ(num_rows, 2);
+    ASSERT_EQ(num_cols, 4);
+    const int num_devices = num_rows * num_cols;
+
+    std::vector<float> test_data;
+    for (int i = 0; i < num_devices; i++) {
+        test_data.insert(test_data.end(), {i * 1.F, i * 2.F, i * 3.F});
+    }
+    Tensor input_tensor =
+        from_vector(test_data, GetTensorSpec(ttnn::SimpleShape{1, num_rows, num_cols, 3}, DataType::FLOAT32));
+
+    auto mapper = api::shard_tensor_2d_to_mesh_mapper(
```

Comment:
what is this `api::` namespace?

Reply:
Yeah, agree, I removed. In general, we use this pattern frequently, which I don't quite understand:

```
namespace ttnn::distributed::api {
// Declare stuff....
}

namespace ttnn::distributed {
using namespace api;
}  
```

I can see how this might be useful to navigate through very large headers, but then would it be better to simply split them up? Other thoughts?